### PR TITLE
Skip activate task in fields system plugin

### PR DESCRIPTION
**Pull request by rdeutz**
Pull Request for Issue #15194

### Summary of Changes
When activating a account we don't need to process the field check/save procedure 

### Testing Instructions
* Add a text field to a user e.g. Company
* Register a user (have set: New user account activation to administrator)
* process the activation

Data get's lost

Apply patch and do the same process

Data will not get lost

### CODE REVIEW

File: plugins/system/fields/fields.php [hunk 1]
```diff
@@ -118,6 +118,14 @@ public function onUserAfterSave($userData, $isNew, $success, $msg)
 
 		$user = JFactory::getUser($userData['id']);
 
+		$task = JFactory::getApplication()->input->get('task');
```
Comment (yvesh) [inline plugins/system/fields/fields.php]: get -> getCmd ? :)

Comment (rdeutz) [inline plugins/system/fields/fields.php]: Change one line and 10 developers have a comment, change 1000 all say great :-)

cmd is the default.

Comment (mbabker) [inline plugins/system/fields/fields.php]: We don't read 1000 lines, we just hope you know what you're doing before you submit the PR 😝

**Comment by AlexRed** [comment]
I have tested this item :white_check_mark: successfully on 8e66e18af66ea73709680b2f8d19c452a464a04a<br /><br />Patch ok for me<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15308">issues.joomla.org/tracker/joomla-cms/15308</a>.</sub>

File: plugins/system/fields/fields.php (hunk 1 above)
Comment (yvesh) [inline plugins/system/fields/fields.php]: @rdeutz i know, but we are already mixing up between getCmd and get .. (most are getCmd, we should be more consistent :P)

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on f6126d39208013d4355e4625350f26d34982026b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15308">issues.joomla.org/tracker/joomla-cms/15308</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
@AlexRed can you please retest?

File: plugins/system/fields/fields.php [hunk 2]
```diff
@@ -118,6 +118,14 @@ public function onUserAfterSave($userData, $isNew, $success, $msg)
 
 		$user = JFactory::getUser($userData['id']);
 
+		$task = JFactory::getApplication()->input->getCmd('task');
+
+		// Skip fields save when we activate a user, because we will lose the saved data
+		if ($task === 'activate')
```
Comment (laoneo) [inline plugins/system/fields/fields.php]: can you add here `|| $task === 'block` as it also doesn't work when Enabling the user

Comment (AlexRed) [inline plugins/system/fields/fields.php]: Yes, I can confirm, block the user empties custom fields

Comment (rdeutz) [inline plugins/system/fields/fields.php]: added block and unblock to the skipped tasks

**Comment by laoneo** [comment]
I guess for now, we are good to go. But this issue revealed a problem which I'm discussing right now with @Bakual to find a proper solution. The we will probably revert that change :speak_no_evil:

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on e15aad7d288b2d38198fb7ddfe2bc392e0e4dc78<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15308">issues.joomla.org/tracker/joomla-cms/15308</a>.</sub>

**Comment by rdeutz** [comment]
> we will probably revert that change 

Never ever I will merge it ;-)

**Comment by Bakual** [comment]
I think for this case the fix is fine anyway. Regardless if we need some handling elsewhere as well.

**Pull request merged**